src/System.IO.FileSystem.DriveInfo/tests/DriveInfo.Windows.Tests.cs

```diff
@@ -0,0 +1,194 @@
+            if (r)
+            {
+                if (fileSystem != validDrive.DriveFormat)
+                {
+                    Assert.False(true, string.Format("Error, unexpected result returned: {0}, Expected: {1}", validDrive.DriveFormat, fileSystem));
+                }
+            }
+            else
+            {
+                Assert.Throws<IOException>(() => { var df = validDrive.DriveFormat; });
+            }
+
+            // Test Invalid drive
+            var invalidDrive = new DriveInfo(GetInvalidDriveLettersOnMachine().First().ToString());
+            Assert.Throws<DriveNotFoundException>(() => { var df = invalidDrive.DriveFormat; });
+        }
+
+        [Fact]
+        public void TestDriveType()
+        {
+            var validDrive = DriveInfo.GetDrives().Where(d => d.DriveType == DriveType.Fixed).First();
+            var expectedDriveType = GetDriveType(validDrive.Name);
+            Assert.Equal((DriveType)expectedDriveType, validDrive.DriveType);
+
+            // Test Invalid drive
+            var invalidDrive = new DriveInfo(GetInvalidDriveLettersOnMachine().First().ToString());
+            Assert.True(invalidDrive.DriveType == DriveType.NoRootDirectory);
+        }
+
+        [OuterLoop]
+        public void TestVolumeLabel()
+        {
+            // Get Volume Label - valid drive
+            var validDrive = DriveInfo.GetDrives().Where(d => d.DriveType == DriveType.Fixed).First();
+            const int volNameLen = 50;
+            StringBuilder volumeName = new StringBuilder(volNameLen);
+            const int fileSystemNameLen = 50;
+            StringBuilder fileSystemName = new StringBuilder(fileSystemNameLen);
+            int serialNumber, maxFileNameLen, fileSystemFlags;
+            bool r = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, out serialNumber, out maxFileNameLen, out fileSystemFlags, fileSystemName, fileSystemNameLen);
+            if (r)
+            {
+                Assert.Equal(volumeName.ToString(), validDrive.VolumeLabel);
+            }
+
+            // Set Volume Label - valid drive
+            string newLabel = "DriveInfoTests";
+            string originalLabel = validDrive.VolumeLabel;
+
+            if (validDrive.IsReady)
+            {
+                bool haveAccess = true;
+                try
+                {
+                    validDrive.VolumeLabel = newLabel;
+                    Assert.Equal(newLabel, validDrive.VolumeLabel);
+                }
+                catch (UnauthorizedAccessException) { haveAccess = false; }
+                finally
+                {
+                    if (haveAccess)
+                        validDrive.VolumeLabel = originalLabel;
+                }
+            }
+
+            // Test UnauthorizedAccess on Network or CD-ROM
+            var noAccessDrive = DriveInfo.GetDrives().Where(d => d.DriveType == DriveType.Network || d.DriveType == DriveType.CDRom);
```

Comment:
I'm not convinced about that for the drive which hasn't been written to yet. It might be a virtual in-memory drive which acts like CD. I just don't want to set/remove someone's label while they are burning CD